t-html/test_demo.py

```diff
@@ -0,0 +1,119 @@
+"""Cover the examples in Andrea's demo."""
+
+from thtml import html, svg
+from thtml.dom import Element, DocumentType
+
+
+def test_automatic_quotes():
+    """Automatic quotes with safe escapes."""
+    fragment = html(t"""<p class={'test & "test"'}>Hello</p>""")
+    assert str(fragment) == '<p class="test &amp; &quot;test&quot;">Hello</p>'
+
+
+def test_sanitized_content():
+    """Sanitized content out of the box"""
+    from random import random
+
+    fragment = html(t"Some random number: {random()}")
+    # QUESTION: Not sure what to test here
+    assert str(fragment).startswith("Some random number")
+
+
+def test_self_closing_tags():
+    """Sanitized content out of the box"""
+    value = "Hello"
+    fragment = html(t"<textarea placeholder={value} />")
+    assert str(fragment) == '<textarea placeholder="Hello"></textarea>'
+
+
+def test_special_attributes():
+    """Data and ARIA"""
+    data = {"a": 1, "b": 2}
+    aria = {"role": "button", "label": "Click me"}
+    fragment = html(t"<div data={data} aria={aria} />")
+    div: Element = fragment.nodes[0]
```

Comment:
it's childNodes now but I wonder why we need to test this directly when all we care about (so far) is the following test as result/outcome